azurerm/internal/services/network/express_route_connection_resource.go

```diff
@@ -0,0 +1,336 @@
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, id.ResourceGroup, id.ExpressRouteGatewayName, id.Name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+		}
+
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return tf.ImportAsExistsError("azurerm_express_route_connection", id.ID())
+		}
+	}
+
+	expressRouteConnectionParameters := network.ExpressRouteConnection{
+		Name: utils.String(id.Name),
+		ExpressRouteConnectionProperties: &network.ExpressRouteConnectionProperties{
+			ExpressRouteCircuitPeering: &network.ExpressRouteCircuitPeeringID{
+				ID: utils.String(d.Get("express_route_circuit_peering_id").(string)),
+			},
+			RoutingWeight: utils.Int32(int32(d.Get("routing_weight").(int))),
+		},
+	}
+
+	if v, ok := d.GetOk("authorization_key"); ok {
+		expressRouteConnectionParameters.ExpressRouteConnectionProperties.AuthorizationKey = utils.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("enable_internet_security"); ok {
+		expressRouteConnectionParameters.ExpressRouteConnectionProperties.EnableInternetSecurity = utils.Bool(v.(bool))
+	}
+
+	if v, ok := d.GetOk("routing"); ok {
+		expressRouteConnectionParameters.ExpressRouteConnectionProperties.RoutingConfiguration = expandExpressRouteConnectionRouting(v.([]interface{}))
+	}
+
+	future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.ExpressRouteGatewayName, id.Name, expressRouteConnectionParameters)
+	if err != nil {
+		return fmt.Errorf("creating/updating %s: %+v", id, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for creation/update of %s: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+
+	return resourceExpressRouteConnectionRead(d, meta)
+}
+
+func resourceExpressRouteConnectionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.ExpressRouteConnectionsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ExpressRouteConnectionID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.ExpressRouteGatewayName, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("express_route_gateway_id", parse.NewExpressRouteGatewayID(id.SubscriptionId, id.ResourceGroup, id.ExpressRouteGatewayName).ID())
+
+	if props := resp.ExpressRouteConnectionProperties; props != nil {
+		d.Set("routing_weight", props.RoutingWeight)
+
+		if v := props.ExpressRouteCircuitPeering; v != nil {
+			d.Set("express_route_circuit_peering_id", v.ID)
+		}
+
+		if v := props.AuthorizationKey; v != nil {
+			d.Set("authorization_key", v)
+		}
+
+		if v := props.EnableInternetSecurity; v != nil {
+			d.Set("enable_internet_security", v)
+		}
```

Comment:
We need to always set these properties, such like 
```suggestion
                peeringID := ""
		if v := props.ExpressRouteCircuitPeering; v != nil {
		        peeringID = *v.ID
		}
			d.Set("express_route_circuit_peering_id", peeringID)

			d.Set("authorization_key", v)

			d.Set("enable_internet_security", v)
```

Reply:
Updated